app/policies/trainee_policy.rb

```diff
@@ -10,7 +10,7 @@ def initialize(user, scope)
     end
 
     def resolve
-      user.system_admin? ? scope.all : scope.where(provider_id: user.provider_id).kept
+      user.system_admin? ? scope.all : scope.where(provider_id: user.primary_provider&.id).kept
```

Comment:
I don't think we should guard against nil provider here. It shouldn't happen and we should raise if it does.